test/LivepeerProtocol.js

```diff
@@ -1192,4 +1197,501 @@ contract('LivepeerProtocol', function(accounts) {
+
+            // Approve token transfer for account 1
+            await lpt.approve(instance.address, a1Stake, {from: accounts[1]});
+
+            // Account 1 bonds to self as transcoder
+            await instance.bond(a1Stake, accounts[1], {from: accounts[1]});
+
+            // Fast forward 1 round
+            await rpc.waitUntilNextBlockMultiple(20, ROUND_LENGTH);
+
+            await instance.initializeRound();
+
+            // Account 2 creates a job
+            await instance.job(1, "0x1", 200, {from: accounts[2]});
+
+            // Account 2 claims work
+            await instance.claimWork(0, 0, 10, "0x1", {from: accounts[1]});
+
+            let threw = false;
+
+            try {
+                // Account 2 claims work again
+                await instance.claimWork(0, 11, 20, "0x2", {from: accounts[1]});
+            } catch (err) {
+                threw = true;
+            }
+
+            assert.isOk(threw, "claim work did not throw when the previous verification is not over");
+        });
+    });
+
+    describe("verify", function() {
+        it("should verify a transcode claim", async function() {
+            const instance = await LivepeerProtocol.new(2, ROUND_LENGTH, CYCLES_PER_ROUND, {from: accounts[0]});
+            const lptaddress = await instance.token.call();
+            const lpt = await LivepeerToken.at(lptaddress);
+
+            const a1Stake = 2000;
+
+            // Transfer tokens
+            await lpt.transfer(accounts[1], toSmallestUnits(1), {from: accounts[0]});
+
+            // Register account 1 as transcoder 1
+            await instance.transcoder(BLOCK_REWARD_CUT, FEE_SHARE, PRICE_PER_SEGMENT, {from: accounts[1]});
+
+            // Approve token transfer for account 1
+            await lpt.approve(instance.address, a1Stake, {from: accounts[1]});
+
+            // Account 1 bonds to self as transcoder
+            await instance.bond(a1Stake, accounts[1], {from: accounts[1]});
+
+            // Fast forward 1 round
+            await rpc.waitUntilNextBlockMultiple(20, ROUND_LENGTH);
+
+            await instance.initializeRound();
+
+            const streamId = 1;
+
+            // Account 2 creates a transcoder job
+            await instance.job(streamId, "0x1", 200, {from: accounts[2]});
+
+            // Segment data hashes
+            const d0 = Buffer.from("80084bf2fba02475726feb2cab2d8215eab14bc6bdd8bfb2c8151257032ecd8b", "hex");
+            const d1 = Buffer.from("b039179a8a4ce2c252aa6f2f25798251c19b75fc1508d9d511a191e0487d64a7", "hex");
+            const d2 = Buffer.from("263ab762270d3b73d3e2cddf9acc893bb6bd41110347e5d5e4bd1d3c128ea90a", "hex");
+            const d3 = Buffer.from("4ce8765e720c576f6f5a34ca380b3de5f0912e6e3cc5355542c363891e54594b", "hex");
+
+            // Segment hashes (streamId, segmentSequenceNumber, dataHash)
+            const s0 = abi.soliditySHA3(["uint256", "uint256", "bytes"], [streamId, 0, d0]);
```

Comment:
Solidity's built-in `sha3` function accepts an arbitrary of number of arguments and tightly packs them together meaning they are concatenated without padding. The `soliditySHA3` function from `ethereumjs-abi` tightly packs arguments before hashing thereby following the behavior of Solidity's `sha3` in Javascript.